# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/actors/BrowserElementParent.jsm
# Commit: edf3df2adebb
# Full Hash: edf3df2adebb5971d458c72078763f3a4f46b9b6
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2019-12-21 21:42:36
# Regressor Bug: 1602410
# File Overlap Count: 3
# Description:
#   Bug 1602410. Make `window.print()` invoke printing via the parent process using a JSWindowActor. r=Gijs,kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56414
# ==============================================================================

diff -r 238918bf6bea -r edf3df2adebb toolkit/actors/BrowserElementParent.jsm
--- a/toolkit/actors/BrowserElementParent.jsm	Sat Dec 21 12:27:40 2019 +0000
+++ b/toolkit/actors/BrowserElementParent.jsm	Sat Dec 21 12:33:11 2019 +0000
@@ -91,6 +91,17 @@
         }
         break;
       }
+      case "ContentRequestedPrint": {
+        // Note: this is not yet fully Fission compatible. That requires
+        // further work to change the PrintUtils API and the underlying code
+        // that it uses.
+        let bc = this.manager.browsingContext;
+        let topWin = bc.top.embedderElement.ownerGlobal;
+        if (topWin.PrintUtils) {
+          topWin.PrintUtils.printWindow(bc);
+        }
+        break;
+      }
     }
   }
 }
